mungegithub/mungers/e2e/resolved.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"sync"
+
+	cache "k8s.io/contrib/mungegithub/mungers/flakesync"
+
+	"github.com/golang/glog"
+)
+
+type resolutionKey struct {
+	job    cache.Job
+	number cache.Number
+}
+
+// ResolutionTracker provides a place for build cops to say "I've resolved this
+// problem" so the merge queue can continue merging.
+type ResolutionTracker struct {
+	resolved map[resolutionKey]bool
+	lock     sync.RWMutex
+}
+
+// NewResolutionTracker constructs an empty resolution tracker. It's the
+// caller's responsibility to hook up Get/SetHTTP.
+func NewResolutionTracker() *ResolutionTracker {
+	return &ResolutionTracker{
+		resolved: map[resolutionKey]bool{},
+	}
+}
+
+// Resolved returns true if the given build has been manually marked as resolved.
+func (r *ResolutionTracker) Resolved(j cache.Job, n cache.Number) bool {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+	return r.resolved[resolutionKey{j, n}]
+}
+
+func getReqKey(req *http.Request) (key resolutionKey, err error) {
+	key.job = cache.Job(req.URL.Query().Get("job"))
+	var n int
+	n, err = strconv.Atoi(req.URL.Query().Get("number"))
+	key.number = cache.Number(n)
+	return key, err
+}
+
+func (r *ResolutionTracker) serve(data interface{}, status int, res http.ResponseWriter) {
+	res.Header().Set("Content-type", "application/json")
+	res.WriteHeader(status)
+	if err := json.NewEncoder(res).Encode(data); err != nil {
+		glog.Errorf("Couldn't write %#v: %v", data, err)
+	}
+}
+
+// GetHTTP accepts "job" and "number" query parameters and returns a json blob
+// indicating whether the specified build has been marked as resolved.
+func (r *ResolutionTracker) GetHTTP(res http.ResponseWriter, req *http.Request) {
```

Comment:
I'd rather we just gave them the whole list and made them parse out the job/key they need. Then again I guess you'd have to find some other way to create the `resolved` map's key so it could be encoded as json...  This way is a black box, no way to know what is in there...